### ISSUE TITLE
Can't see data(scalars/images) in tensorboard

### ISSUE THREAD

**Issue by BostonLobster** [MIC-DKFZ/trixi#22]
I use the develop branch of trixi and install it by `pip install -e TRIXI_DIR`.

I use the `PytorchExperiment` infrastructure and use `self.add_result` to add scalars into tensorboard logger.

The `tfevents` files indeed exist in the save/tensorboard folder, but they are not shown in the tensorbard webpage.

How can I inspect this problem?

**Comment by BostonLobster** [comment]
After pulling to the newest commit on develop branch, and running the same experiments still fails to see data in tensorbard webpage...

Where is wrong?

**Comment by jenspetersen** [comment]
Is tensorboard pointed to the right directory? Here's a minimal example that works for me:

    from trixi.experiment import PytorchExperiment
    exp = PytorchExperiment(base_dir="some_directory", loggers={"tb": "tensorboard"}
    exp.add_result(1, "test")

Maybe you can try that too? Note that if you don't specify any arguments for the TensorboardXLogger (still named this way because we used to use tensorboardx instead of the PyTorch integration), the output, i.e. tfevents and such, will be found in the save directory of your experiment. You can also specify another directory for the logger like this:

    ..., loggers={"tb": ("tensorboard", {"target_dir": "some_other_directory"})}

Could it be that you're just pointing the tensorboard server to the wrong directory?
If that isn't the problem, please construct a minimal example where it doesn't work and also tell us a little more about your environment. pip freeze would help.

**Comment by BostonLobster** [comment]
Hi, Jens. Running your minimal experiment succeeds. 

My `target_dir` is `None`, so tensorboard should points to elog dir. And there are indeed tfevents files in `EXP_DIR/save/tensorboard`. 

But when I run tensorboard with logdir pointed to this EXP_DIR, there is still no data in webpage...

Moreover, when I run `tensorboard --inspect --logdir EXP_DIR`, it shows the following:
> TensorFlow installation not found - running with reduced feature set.======================================================================
Processing event files... (this can take a few minutes)
======================================================================
Found event files in:
./save/tensorboard
./save/tensorboard/TV Loss/TV Loss
./save/tensorboard/Landmark Loss/Landmark Loss
These tags are in ./save/tensorboard:
audio -
histograms -
images
   test
scalars -
tensor -
======================================================================
Event statistics for ./save/tensorboard:
audio -
graph -
histograms -
images
   first_step           1
   last_step            19001
   max_step             19001
   min_step             1
   num_steps            20
   outoforder_steps     []
scalars -
sessionlog:checkpoint -
sessionlog:start -
sessionlog:stop -
tensor -
======================================================================
These tags are in ./save/tensorboard/TV Loss/TV Loss:
audio -
histograms -
images -
scalars
   TV_Loss
tensor -
======================================================================
Event statistics for ./save/tensorboard/TV Loss/TV Loss:
audio -
graph -
histograms -
images -
scalars
   first_step           0
   last_step            0
   max_step             0
   min_step             0
   num_steps            1
   outoforder_steps     []
sessionlog:checkpoint -
sessionlog:start -
sessionlog:stop -
tensor -
======================================================================
These tags are in ./save/tensorboard/Landmark Loss/Landmark Loss:
audio -
histograms -
images -
scalars
   Landmark_Loss
tensor -
======================================================================
Event statistics for ./save/tensorboard/Landmark Loss/Landmark Loss:
audio -
graph -
histograms -
images -
scalars
   first_step           0
   last_step            0
   max_step             0
   min_step             0
   num_steps            1
   outoforder_steps     []
sessionlog:checkpoint -
sessionlog:start -
sessionlog:stop -
tensor -
======================================================================

Any suggestion?

**Comment by jenspetersen** [comment]
Well the inspection shows that there are scalars for both TV Loss and Landmark Loss. If you do

    exp.add_result(0.1, "TV Loss", counter=0)

in my example, the inspection will give the same output. Maybe restart tensorboard, refresh the page? I know it sounds obvious, but everything looks okay, so maybe it's some weird caching issue with your browser? Also note that the scalar outputs are collapsed by default. Or does tensorboard actually say "No scalar data was found."? Is the image data displayed correctly? I'm also not sure how tensorboard handles NaN, maybe that's your issue?

**Comment by BostonLobster** [comment]
@jenspetersen Thanks for your reply!

I conducted an ablation study on this, and I have the following obervations:
1. If I change the folder_format to `'{experiment_name}_%Y%m%d-%H:%M:%S'` (this is which I used when creating this issue), both the scalar and images are not shown. (regardless of how I refresh webpage or restart tensorboard)
2. If I remove the `:` in folder_format, the images are shown immediately, and the scalars will present after I restart tensorboard several times.
OK, maybe the colon in folder_format will be one cause. Let's continue...
3. Since I initially use importlib to import Experiment class, like this
```
module = import_module('experiments.{}'.format(c.exp_name))
exp = module.Experiment()
```
Could it be a cause?
After adding it, the images show perfectly, and the scalars appear after a few seconds(don't know why?) So importlib isn't a disrupter.

Conclusion:
The colon is the culprit! But... Can it be allowed? I saw other projects also use colon in log_dir name.

Anyway, many many thanks to your kind suggestions!
